src/skmatter/metrics/_pairwise.py

```diff
@@ -0,0 +1,213 @@
+from typing import Union
+
+import numpy as np
+from sklearn.metrics.pairwise import (
+    _euclidean_distances,
+    check_array,
+    check_pairwise_arrays,
+)
+
+
+def pairwise_euclidean_distances(
+    X, Y=None, *, Y_norm_squared=None, squared=False, X_norm_squared=None, cell=None
```

Comment:
Also check here the correct shape of the input you expect. Especially for the cell because there are a lot of different definitions.